vsphere/datadog_checks/vsphere/data/conf.yaml.example

```diff
@@ -82,49 +60,63 @@ instances:
-    ## not to be slowed down by the collection of historical metrics.
-    ## Note: An instance with this flag set to true does not submit any service check, events or host tags.
-    ##       It is designed to be used only when next to another instance of the check.
-    #
-    # collect_historical_only: false
+    ## @param resource_filters - list of objects - optional - default: no filter
+    ## Each filter in the list is composed of three parameters.
+    ## 'resource' is one of vm/host/datastore/datacenter/cluster on which to apply the filter
+    ## 'patterns' is a list of matching regex, if any of those matches, the resource will be monitored.
+    ## 'property' is the kind of property on which to apply the filter and must be one of those values:
+    ##    - 'property: name' (default) to filter on the resource name (defined by vCenter)
+    ##    - 'property: inventory_path' to filter on the full inventory path
+    ##       (ex: \/<Datacenter_name>\/vm\/<FOLDER_NAME>\/.* for all vms in the folder <FOLDER_NAME>)
+    ##    - 'property: hostname' to filter on the ESXi Host name (only valid for if 'resource' equals 'vm')
+    ##    - 'property: guest_hostname' to filter on the VM hostname (only valid for if 'resource' equals 'vm' and
+    ##       if VMware tools are installed)
+    ## Note1: 'datastore', 'datacenter' and 'cluster' filters are ignored when collecting realtime metrics as those
+    ##         resources do not have realtime metrics
+    ## Note2: 'vm' and 'host' filters are ignored when collecting historical metrics as those resources do not have
+    ##         historical metrics
+    #
+    # resource_filters:
+    #   - resource: vm
+    #     property: name
+    #     patterns:
+    #       - <VM_REGEX_1>
+    #       - <VM_REGEX_2>
+    #   - resource: vm
+    #     property: hostname
+    #     patterns:
+    #       - <HOSTNAME_REGEX>
+    #   - resource: vm
+    #     property: guest_hostname
+    #     patterns:
+    #       - <GUEST_HOSTNAME_REGEX>
+    #   - resource: host
+    #     property: inventory_path
+    #     patterns:
+    #       - <INVENTORY_PATH_REGEX>
+
+    ## @param metric_filters - object - optional - default: all metrics
+    ## For each resource type (vm, host, datastore, datacenter, cluster) collected,
+    ## you can choose which metric you want to collect using a list of regex.
+    ## If you do not specify a regex for the resource, all metrics will be collected.
+    ## See https://github.com/DataDog/integrations-core/blob/master/vsphere/metadata.csv
+    ## for the list of collected metrics.
+    ## Note1: 'datastore', 'datacenter' and 'cluster' filters are ignored when collecting realtime metrics as those
+    ##         resources do not have realtime metrics
+    ## Note2: 'vm' and 'host' filters are ignored when collecting historical metrics as those resources do no
+    ##         have historical metrics
+    #
+    # metric_filters:
```

Comment:
I'm not sure if there is a need to separate by resource type here. This would do IMO:

```
metric_filters:
  - <REGEX_PATTERN>
  - <REGEX_PATTERN>
  - <REGEX_PATTERN>
```

The metrics are distinct enough.

That will also allow user to more easily exclude a pattern across resource type. 
- e.g. `.*\.cpu\..*` to exclude all cpu things.
- e.g. `.*\.avg` to exclude all .avg things.

Reply:
Not possible :) 

a single metric (let's say `vsphere.cpu.usage.avg`) is available for VMs, Hosts and Clusters.
You need to have a way to say "I want this metric for VMs but not for Clusters". Especially knowing that there's a good amount of Cluster-level metric that are just aggregate and are usually not very helpful (but hard to query because historical)

Reply:
Make sense if a single metric might have multiple resource types.